src/lib/transaction_snark/transaction_snark.ml

```diff
@@ -3644,6 +3644,7 @@ let verify (ts : (t * _) list) ~key =
       (module Statement.With_sok)
       key
       (List.map ts ~f:(fun ({ statement; proof }, _) -> (statement, proof)))
+    |> Promise.to_deferred
```

Comment:
Can you rename `Pickles.verify` to `Pickles.verify_promise`, and then add a deferred `Pickles.verify` version to use here and elsewhere?

Reply:
Sure!

Reply:
Done